modules/rostests/kmtests/ntos_ex/ExUuid.c

```diff
@@ -14,8 +15,32 @@ START_TEST(ExUuid)
 {
     UUID Uuid;
     NTSTATUS Status;
+    unsigned char V;
 
     Status = ExUuidCreate(&Uuid);
+    // FIXME: Why does Test WHS report RPC_NT_UUID_LOCAL_ONLY instead?
     ok(Status == STATUS_SUCCESS, "ExUuidCreate returned unexpected status: %lx\n", Status);
-    ok((Uuid.Data3 & 0x1000) == 0x1000, "Invalid UUID version: %x\n", (Uuid.Data3 & 0xF000));
+
+    V = (Uuid.Data3 & 0xF000) >> 12;
+    ok(V == 1, "UUID version: expected 1, got %u\n", V);
+
+    V = (Uuid.Data4[0] & 0xF0) >> 4;
+    if ((V & 0b1000) == 0b0000)
+    {
+        V = 0;
+    }
+    else if ((V & 0b1100) == 0b1000)
+    {
+        V = 1;
+    }
+    else if ((V & 0b1110) == 0b1100)
+    {
+        V = 2;
+    }
+    else // if ((V & 0b1110) == 0b1110)
+    {
+        // Reserved. Fake a value.
+        V = 255;
+    }
+    ok(V == 1, "UUID variant: expected 1, got %u\n", V);
```

Comment:
This seems a bit excessive. Failure is not expected, so we don't need it particularly user-friendly. I think as a rule, it's good to handle failure safely (e.g. exit on allocation failure) and actionably (e.g. print out the failing values so that somebody can compare and investigate).
Am I understanding correctly that simply changing the original check to `(Uuid.Data3 & 0xF000) == 0x1000` would address this issue (while still adhering to these goals)? If so, I would prefer that solution to adding this much extra code.

Reply:
I understand what you wrote, but it does not seem to apply in this context.
Did you miss that this second part is about 'variant', not 'version'?

Reply:
Okay, so you're adding an additional test, too. The same applies there -- make it one line.

Reply:
> one line

Done.